src/main/java/bio/overture/ego/security/OAuth2SsoFilter.java

```diff
@@ -61,6 +69,53 @@ public void onAuthenticationSuccess(
   private OrcidService orcidService;
   private GithubService githubService;
 
+  private AuthenticationFailureHandler failureHandler =
+      (request, response, exception) -> {
+        val rootExceptionThrowable = exception.getCause();
+        val application =
+            applicationService.getByClientId(
+                (String) request.getSession().getAttribute("ego_client_id"));
```

Comment:
Is it possible that we have a user not an application attempting to sign in and get caught up in this error handler? What happens in all the application and redirect specific code here when its a user with no client id or redirect uri?

Reply:
Ignore this comment, I forgot that User logins occur through applications so they all require an application client id in the request. No issues here.